### ISSUE TITLE
ERROR LOADING PYREVIT (4.8.5)

### ISSUE THREAD

**Issue by RDSalvi** [eirannejad/pyRevit#1163]
I did the installation of pyrevit 4.8.5, it occurred normally.
When I start the Revit 2020 software, the pyrevit plugin loading error appears

Below are screenshots of the errors

![2021-02-09_11h09_41](https://user-images.githubusercontent.com/78804550/107375604-b0ab8900-6ac7-11eb-96c8-3c5c98b2efcc.png)
![2021-02-09_11h10_09](https://user-images.githubusercontent.com/78804550/107375610-b1dcb600-6ac7-11eb-84c2-7e0f47bf2a15.png)
![2021-02-09_11h10_25](https://user-images.githubusercontent.com/78804550/107375612-b2754c80-6ac7-11eb-9adc-1dc00e6d3ad7.png)

**Comment by eirannejad** [comment]
Would you mind going into the `%APPDATA%/pyRevit/2020` folder and share the log files?

**Comment by RDSalvi** [comment]
![2021-02-09_13h24_06](https://user-images.githubusercontent.com/78804550/107394099-405a3300-6ada-11eb-8479-f96995e976ac.png)
![2021-02-09_13h24_21](https://user-images.githubusercontent.com/78804550/107394102-40f2c980-6ada-11eb-8373-6d476898cbc7.png)

**Comment by eirannejad** [comment]
Huh. So this bug happens way early in the load process. Okay. Open the `pyRevit_config.ini` file and set **File Logging** to **True** please. Then run again. You should see a `2020` folder with a log file inside it after the crash

**Comment by camozpod67** [comment]
Hello, I am getting a similar message when loading pyRevit in multiple versions of Revit. However, I am getting a log output. I am getting a similar output for each version of Revit. This is the one from Revit 2019.2.2:
`2021-02-11 08:07:49,347 ERROR [pyrevit.userconfig] Can not determine cpython engines for current attachment: Unknown | Product: "2019.2.2 Hotfix" | Manifest: "C:\Users\p006031D\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
`
Not sure if this is the same issue so let me know if I should open a separate issue for this. Thanks.

**Comment by yongjoon-kim** [comment]
Hello @eirannejad, is there any update on this issue? I'm getting the same error message.

**Comment by eirannejad** [comment]
@yongjoon-kim Honestly I don't have enough info to understand the root cause of the error. On the surface it seems like a dll conflict with another Revit addon. Do you have anything else installed that might ship the `Microsoft.CodeAnalysis.CSharp.dll` with it?

**Comment by yongjoon-kim** [comment]
@eirannejad Hi Ehsan, thanks for the response! So here's what I found out:

1. I've been having this Revit issue which prevented me from loading the built-in Revit add-ins such as `Collaborate.dll` or things like that.
2. I spoke with Autodesk team and found out that it happens when the Windows user profile is corrupt. They can't figure our the actual reason behind it, but I assume this is related to an incomplete Windows update that might have harmed the .net 4.8 framework on the local machine
3. Once I fixed this issue by setting up a new Windows user profile, it started to load pyRevit without any issue. So I guess you were right, it's related to the conflict in the add-in assembly versions.

Long story short, it's most likely the problem of a local machine, not the pyRevit itself. Good lessons learned here!

**Comment by eirannejad** [comment]
@yongjoon-kim Awesome! thank you for following through with this and keeping me updated! 👍

**Comment by DharmanGersch** [comment]
I've had the same issue today after installing Pyrevit 4.8.8. I removed the https://github.com/BHoM addin and it works now. BHoM also had `Microsoft.CodeAnalysis.CSharp.dll` which i'm assuming caused a conflict. This is the version i removed from BHoM 

![image](https://user-images.githubusercontent.com/10596012/127606408-441812ab-2a86-44b6-860a-32a870ddad8c.png)

**Comment by Acamarvin** [comment]
Hi!

I also did the installation of Pyrevit but 4.8.9.21361+0320 version , also normally.
But, when starting Revit 2021, the Pyrevit plugin loading error appears.

I'm very new using commands, in fact, I never use them, but I need your help, please.... Could you be very patient and help me with the step by step to fix it and start usin Pyrevit please????

Below the screenshots' errors....

Thanks
![Screenshot (20220130-005457)](https://user-images.githubusercontent.com/98682670/151690473-9d4fa86e-e185-40d5-a51e-c849572da774.jpg)
![Screenshot (20220130-005523)](https://user-images.githubusercontent.com/98682670/151690476-91a745a1-392f-4a74-8f1b-9a371ed246ed.jpg)
![Screenshot (20220130-005654)](https://user-images.githubusercontent.com/98682670/151690484-8409bdb0-d384-4a5f-a894-b7cb4cb7c9ea.jpg)
!

**Comment by eirannejad** [comment]
Similar to https://github.com/eirannejad/pyRevit/issues/1451

**Comment by Acamarvin** [comment]
Thanks for tour answer... But some houra ago I think I fix It,.... How was it??? Ok... I copied the pyrevit_config.ini file from the ProgramData folder (where it is originally located the máster file) to the pyrevit folder in AppData folder (where, as you saw in the screenshors, it is missing).

Then, after this, I restarted the PC and when opening again Revit, the pyrevit loading error message doesn't appeard any more. So, It seems Pyrevit plugin working perfect after testing it... I hope it was fixed definetly. What do you think???? Could this bring troubles someday???

Note: Also, never appeared the "always load Pyrevit" dialogue box when launching Revit, is It ok???? But, I said before, It seems working well for the moment.

I wait for your comments... And thanks for answering!!!

**Comment by reutben9** [comment]
Hi, I also have this problem. I don't know anything in coding, but I get that there might be an issue with Microsoft.CodeAnalysis.CSharp.dll
So I searched for it and both pyrevit and enscape have it. I don't want to delete any of them. What should I do?

**Comment by pflondrigan** [comment]
Hi, I am having a similar issue. However, Revit is telling me that it failed to initialize the add-in "PyRevitLoader" because the assembly "%APPDATA%\pyRevitLoader.dll" does not exist.  Everything has been installed per the manufacturer, multiple times.  The only other plugin that I have on Revit is Enscape. 

Thanks!
